addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -509,6 +509,7 @@ const LinkTools = Link.extend({
         // Without this, no update if input is same as original text.
         this._updateLinkContent(this.$link, data, {force: true});
         this._observer.observe(this._link, this._observerOptions);
+        this.options.wysiwyg.odooEditor.historyStep();
```

Comment:
Shouldn't the change be caught automatically by the observer anyway? There is an automatic `historyStep` in `observerActive`:
```js
                if (this._observerTimeoutUnactive.size === 0) {
                    this.observerTimeout = setTimeout(() => {
                        this.historyStep();
                    }, 100);
                }
```

Reply:
@dmo-odoo , `historyStep` is not called here because the linkZws mechanism used to call `observerUnactive` which clears `observerTimout`, after the merge of the new link mechanism (https://github.com/odoo/odoo/pull/162575), it's fixed  so I'm closing this PR

Reply:
@dmo-odoo  , I updated the fix and I explained the whole flow , I hope it's clear now